src/main/java/edu/ucdavis/dss/ipa/api/components/scheduling/SchedulingViewController.java

```diff
@@ -127,4 +127,23 @@ public Activity createActivity(@PathVariable char activityCode, @PathVariable Lo
 		newActivity = activityService.saveActivity(newActivity);
 		return newActivity;
 	}
+
+	@RequestMapping(value = "/api/schedulingView/sections/{sectionId}/activities", method = RequestMethod.GET, produces="application/json")
+	@ResponseBody
+	public List<Activity> getActivitiesBySection(@PathVariable Long sectionId, HttpServletResponse httpResponse) {
+		Section section = sectionService.getOneById(sectionId);
+
+		if (section == null) {
+			httpResponse.setStatus(HttpStatus.BAD_REQUEST.value());
+			return null;
+		}
+
+		Authorizer.hasWorkgroupRole(section.getSectionGroup().getCourse().getSchedule().getWorkgroup().getId(), "academicPlanner");
```

Comment:
Not a comment related to your work here but yikes (!) with the .get().get().get().get()